File: src/test/java/br/com/rcc_dev/testes/mocks/CepControlMock.java
package br.com.rcc_dev.testes.mocks;

import br.com.rcc_dev.testes.controls.CepControl;
import br.com.rcc_dev.testes.entities.CepOutput;
import br.com.rcc_dev.testes.entities.database.Cep;
import io.vertx.core.Future;

public class CepControlMock extends CepControl {
  
  public Future<CepOutput> findCEP(String cep){
    CepOutput obj = new CepOutput();
  
    obj.setCep(cep);
    obj.setCidade("Cidade aqui");
    obj.setBairro("Bairro aqui");
    obj.setId(1);
    
    return Future.succeededFuture(obj);
  }
  
}

File: src/test/kotlin/br/com/rcc_dev/testes/mocks/CepControlMock.kt
package br.com.rcc_dev.testes.mocks

import br.com.rcc_dev.testes.controls.CepControl
import br.com.rcc_dev.testes.entities.Cep
import io.vertx.core.Future

class CepControlMock: CepControl() {

  override fun findCEP(cep: String): Future<Cep> {
    val obj = Cep(
        id =      1 ,
        cep =     cep ,
        cidade =  "Cidade aqui" ,
        bairro =  "Bairro aqui"
    )

    return Future.succeededFuture<Cep>(obj)
  }

}